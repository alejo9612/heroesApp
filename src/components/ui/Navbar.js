import React, { useContext } from 'react';
import { Link, NavLink, useHistory } from 'react-router-dom';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const Navbar = () => {

    //llamamos el useContext y lo desestructuramos, también desestructuramos lo que trae la primera desestructuracion
    const {user:{name}, dispatch} = useContext(AuthContext);
    //console.log(name)

    //Este es un hook del react ruter el cual nos provee la información que necesitamos del history cuando la requerimos
    const history= useHistory();

    const handlelogout = () =>{
        
        history.replace('/login');
        //me dispara de nuevo el state del reducer con los valores iniciales, deslogando a mi usuario
        dispatch({
            type: types.logout
            //name
        });

        
    }

    return (

        <nav className="navbar navbar-expand-sm navbar-dark bg-dark container-fluid">
            
            <Link 
                className="navbar-brand" 
                to="/"
            >
                <img src={`../assets/heroes/marvel-icono.png`} 
                     style={{maxWidth:100}}
                     alt="Marvel"
                />
            </Link>

            <div className="navbar-collapse">
                <div className="navbar-nav">

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/marvel"
                    >
                        <b>Marvel</b>
                    </NavLink>

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/dc"
                    >
                        <b>DC</b>
                    </NavLink>

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/Buscar"
                    >
                        <b>Buscar</b>
                    </NavLink>
                </div>
            </div>

            <div className="d-flex collapse order-3 dual-collapse2">
                <ul className="navbar-nav ml-auto">
                    <span className="nav-link text-info">
                        {name}
                    </span>

                    <button className="nav-link nav-link btn btn-outline-dark"
                            onClick={handlelogout}
                    >
                        Logout
                    </button>
                </ul>
            </div>
        </nav>
    )
}