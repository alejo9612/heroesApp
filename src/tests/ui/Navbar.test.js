import React from "react";
import { mount } from "enzyme";
import { MemoryRouter, Router } from "react-router-dom";
import { AuthContext } from "../../auth/AuthContext";
import { Navbar } from '../../components/ui/Navbar';
import { types } from '../../types/types';

describe('Se llevar√°n a cabo las pruebas del <Navbar/>', () => {

    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    }

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            name: 'Camila',
            logged: true
        }
    }

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <Navbar />
                </Router>
            </MemoryRouter>
        </AuthContext.Provider>
    );

    afterEach(() => {
        jest.clearAllMocks();
    });

    // console.log(wrapper.html());

    test('Debe de mostrarse de manera correcta ', () => {

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('Camila');

    });

    test('Debe de llamar el logout y usar el history', () => {

        //simulamos de esta manera el uso del click
        wrapper.find('button').prop('onClick')();

        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.logout
        });

    });

});
