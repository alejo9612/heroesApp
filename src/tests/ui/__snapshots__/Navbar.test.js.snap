// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Se llevar√°n a cabo las pruebas del <Navbar/> Debe de mostrarse de manera correcta  1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark container-fluid"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    <img
      alt="Marvel"
      src="./assets/heroes/marvel-icono.png"
      style={
        Object {
          "maxWidth": 100,
        }
      }
    />
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        <b>
          Marvel
        </b>
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        <b>
          DC
        </b>
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        <b>
          Buscar
        </b>
      </a>
    </div>
  </div>
  <div
    className="d-flex collapse order-3 dual-collapse2"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-link text-info"
      >
        Camila
      </span>
      <button
        className="nav-link nav-link btn btn-outline-dark"
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
